lua/xcodebuild/project/manager.lua

```diff
@@ -41,6 +41,58 @@ local function validate_xcodeproj_tool()
   return true
 end
 
+---Searches for a `.xcodeproj` file in the given directory.
+---
+---Uses the `find` command to search for the first `.xcodeproj` file
+---in the specified directory and returns its absolute path if found.
+---
+---@param dir string The directory to search in.
+---@return string|nil The absolute path to the `.xcodeproj` file, or nil if not found.
+local function find_xcodeproj_file(dir)
+  local function shell_escape(str)
+    return "'" .. string.gsub(str, "'", "'\\''") .. "'"
+  end
+
+  local cmd = "find " .. shell_escape(dir) .. ' -maxdepth 1 -name "*.xcodeproj" -print -quit 2> /dev/null'
+  local result = util.shell(cmd)
+  if result and #result > 0 and result[1] ~= "" then
+    return result[1]
+  else
+    return nil
+  end
+end
+
+---Traverses up the directory path to find a `.xcodeproj` file.
+---
+---Starting from the given path, this function moves up the directory
+---hierarchy searching for a `.xcodeproj` file in each directory.
+---Returns the absolute path to the `.xcodeproj` file if found.
```

Comment:
About cwd, do you think it's plausible it might start searching from a directory outside cwd? Suppose cwd is /Users/me/project1 and the file being added is /Users/me/project2/some/file.swift. In that case `dir ~= cwd` will never trigger

Reply:
Hmm, I think we can skip that case, but I would ensure this loop won't get stuck, so I would also stop if we hit `/` also please try this case to make sure it works.

Reply:
I've tried and it doesn't get stuck since at the end of each iteration it removes trailing `/` from a string which turns "/" into "", but I'll add an additional condition anyway just to be explicit about it

Reply:
Hmm it seems like in Lua `""` evaluates to true anyway, but for some reason I don't get stuck in a loop.
I'll add an additional condition for empty string too

Reply:
I'm thinking about one more thing. Maybe it would be better to use built-in function instead of regex:
```
file_path = vim.fn.fnamemodify(file_path, ':h')
```

I think it would be easier to read and it stops on `/` - it won't produce an empty string for an absolute path.

Reply:
Oh yeah, you're totally right, I didn't know that was a thing